src/shell/scripts/omp.ps1

```diff
@@ -120,115 +134,128 @@ New-Module -Name "oh-my-posh-core" -ScriptBlock {
         return $pswd
     }
 
-    if (("::TOOLTIPS::" -eq "true") -and ($ExecutionContext.SessionState.LanguageMode -ne "ConstrainedLanguage")) {
-        Set-PSReadLineKeyHandler -Key Spacebar -BriefDescription 'OhMyPoshSpaceKeyHandler' -ScriptBlock {
-            [Microsoft.PowerShell.PSConsoleReadLine]::Insert(' ')
-            $command = $null
-            [Microsoft.PowerShell.PSConsoleReadLine]::GetBufferState([ref]$command, [ref]$null)
-            $command = $command.TrimStart().Split(" ", 2) | Select-Object -First 1
-            # ignore an empty/repeated tip
-            if ($command -eq '' -or $command -eq $script:ToolTipCommand) {
-                return
+    function Get-TerminalWidth {
+        $terminalWidth = $Host.UI.RawUI.WindowSize.Width
+        # Set a sane default when the value can't be retrieved.
+        if (-not $terminalWidth) {
+            return 0
+        }
+        $terminalWidth
+    }
+
+    function Update-PoshRPrompt {
+        param([string]$NewRPrompt)
+
+        $position = $Host.UI.RawUI.CursorPosition
+        $terminalWidth = Get-TerminalWidth
+        if ($NewRPrompt) {
+            # Save the right/tooltip prompt to avoid unnecessary CLI calls.
+            $script:CurrentRPrompt = $NewRPrompt.TrimStart()
+            $paddingLength = $NewRPrompt.Length - $script:CurrentRPrompt.Length
```

Comment:
These prompts contain ANSI characters, how is the length calculation correct?

Reply:
Just as the code does. The prompt rendered by the OMP executable always contains leading spaces as padding for right-alignment, so we can calculate *padding_length*, and the **real** tooltip/right prompt length is *terminal_length* - *cursor_position_X* - *padding_length*, where counting ANSI characters is avoided.